netplan_cli/cli/commands/status.py

```diff
@@ -109,184 +255,613 @@ def pretty_print(self, data: JSON, total: int, _console_width=None) -> None:
+            if self.ifname and self.ifname != interface:
+                continue
+
+            self.pprint('{sign} [{col}]● {idx:>2}  {name} {type} {state}[/{col}]'.format(
+                sign=sign,
+                col='red',
+                idx='',
+                name=interface,
+                type=properties.get('type'),
+                state='MISSING',
+                ))
+            self.pprint()
+
+    def plain_print(self, *args, **kwargs):
+        if len(args):
+            lst = list(args)
+            for tag in MATCH_TAGS.findall(lst[0]):
+                # remove matching opening and closing tag
+                lst[0] = lst[0].replace('[{}]'.format(tag), '')\
+                               .replace('[/{}]'.format(tag), '')
+            return print(*lst, **kwargs)
+        return print(*args, **kwargs)
 
-            pprint()
+    def pretty_print(self, data: JSON, total: int, _console_width=None) -> None:
+        self.pprint = self._create_pretty_print(_console_width)
+        self.PLUS = '[green]+[/green]'
+        self.MINUS = '[red]-[/red]'
+        self.PAD = '18'
+        if self.diff:
+            # In diff mode we shift the text 2 columns to the right so we can display
+            # + and - and maintain the alignment consistency
+            self.PAD = '20'
+
+        # Global state
+        if not self.diff:
+            self._display_global_state(data)
+
+        # Per interface
+        interfaces = [(key, data[key]) for key in data if key != 'netplan-global-state']
+        for (ifname, ifconfig) in interfaces:
+            # If we called netplan status for a single interface, ignore the rest
+            if self.ifname and self.ifname != ifname:
+                continue
+
+            self._display_interface_header(ifname, ifconfig)
+            self._display_mac_address(ifname, ifconfig)
+            self._display_ip_addresses(ifname, ifconfig)
+            self._display_dns_addresses(ifname, ifconfig)
+            self._display_dns_search(ifname, ifconfig)
+            self._display_routes(ifname, ifconfig)
+            self._display_bridge(ifname, ifconfig)
+            self._display_bond(ifname, ifconfig)
+            self._display_vrf(ifname, ifconfig)
+            self._display_members(ifname, ifconfig)
+            self._display_activation_mode(ifconfig)
+
+            if not self.diff_only or self._has_diff(ifname):
+                self.pprint()
+
+        if self.diff:
+            self._display_missing_interfaces()
 
         hidden = total - len(interfaces)
         if (hidden > 0):
-            pprint('{} inactive interfaces hidden. Use "--all" to show all.'.format(hidden))
+            self.pprint('{} inactive interfaces hidden. Use "--all" to show all.'.format(hidden))
+
+        if (self.diff and not self.diff_only):
+            self.pprint('Use "--diff-only" to omit the information that is consistent between the system and Netplan.')
```

Comment:
nitpick: This looks a bit odd, due to the `'netplan.str': 'grey62',` definition in the diff Theme. Can we make this line to use the usual theme?

Reply:
To use the other theme I'd need to create a new Console() so I guess I'll just add [yellow] around the --diff-only. I think it's the only part being affected by the theme...